### DISCUSSION TITLE
roslaunch error on the drone_control package

### DISCUSSION THREAD

**Discussion by BKC92** [SMRazaRizvi96/MR-based-drone-controller#3] [Q&A]
Hi Raza

I tried to test if roslaunch works well before connecting the tello drone, but I got this error.
It's the first time I've connected a drone and a HoloLens, so I don't know if it works normally if I connect the drone before roslaunch, or if I'm following your setup well. 
Any help would be appreciated.

Thanks

user@user-VirtualBox:~/catkin_ws$ roslaunch drone_control joy_node.launch
... logging to /home/user/.ros/log/a4a57506-e641-11ed-b8fc-5390d9853ada/roslaunch-user-VirtualBox-5681.log
Checking log directory for disk usage. This may take a while.
Press Ctrl-C to interrupt
Done checking log file disk usage. Usage is <1GB.

started roslaunch server http://user-VirtualBox:46475/

SUMMARY
========

PARAMETERS
 * /aruco_single/camera_frame: stereo_gazebo_lef...
 * /aruco_single/corner_refinement: LINES
 * /aruco_single/image_is_rectified: True
 * /aruco_single/marker_frame: aruco_marker_frame
 * /aruco_single/marker_id: 582
 * /aruco_single/marker_size: 0.034
 * /aruco_single/reference_frame: 
 * /joy_node/dev: /dev/input/js6
 * /rosdistro: noetic
 * /rosversion: 1.16.0
 * /tello/image_raw/image_transport: h264
 * /tello/new_tello_driver_node/camera_calibration: /home/user/catki...
 * /tello/new_tello_driver_node/connect_timeout_sec: 10.0
 * /tello/new_tello_driver_node/local_cmd_client_port: 8890
 * /tello/new_tello_driver_node/local_vid_server_port: 6038
 * /tello/new_tello_driver_node/stream_h264_video: True
 * /tello/new_tello_driver_node/tello_cmd_server_port: 8889
 * /tello/new_tello_driver_node/tello_ip: 192.168.10.1

NODES
  /
    aruco_single (aruco_ros/single)
    joy_node (joy/joy_node)
    tello_NEW_Aruco_Pose (drone_control/tello_NEW_Aruco_Pose.py)
    tello_joypad_control (drone_control/tello_joypad_control.py)
  /tello/
    image_raw (image_transport/republish)
    new_tello_driver_node (drone_control/new_tello_driver_node.py)

auto-starting new master
process[master]: started with pid [5691]
ROS_MASTER_URI=http://localhost:11311

setting /run_id to a4a57506-e641-11ed-b8fc-5390d9853ada
process[rosout-1]: started with pid [5701]
started core service [/rosout]
process[aruco_single-2]: started with pid [5704]
ERROR: cannot launch node of type [drone_control/new_tello_driver_node.py]: Cannot locate node of type [new_tello_driver_node.py] in package [drone_control]. Make sure file exists in package path and permission is set to executable (chmod +x)
process[tello/image_raw-4]: started with pid [5705]
ERROR: cannot launch node of type [drone_control/tello_NEW_Aruco_Pose.py]: Cannot locate node of type [tello_NEW_Aruco_Pose.py] in package [drone_control]. Make sure file exists in package path and permission is set to executable (chmod +x)
ERROR: cannot launch node of type [joy/joy_node]: joy
ROS path [0]=/opt/ros/noetic/share/ros
ROS path [1]=/home/user/catkin_ws/src
ROS path [2]=/opt/ros/noetic/share
ERROR: cannot launch node of type [drone_control/tello_joypad_control.py]: Cannot locate node of type [tello_joypad_control.py] in package [drone_control]. Make sure file exists in package path and permission is set to executable (chmod +x)
[ WARN] [1682740513.390417153]: Corner refinement options have been removed in ArUco 3.0.0, corner_refinement ROS parameter is deprecated
[ WARN] [1682740513.435881874]: normalizeImageIllumination is unimplemented!

**Comment by SMRazaRizvi96** [comment]
To me, the error:

ERROR: cannot launch node of type [drone_control/new_tello_driver_node.py]: Cannot locate node of type [new_tello_driver_node.py] in package [drone_control]. Make sure file exists in package path and permission is set to executable (chmod +x)

Seems to be an error of the ROS executor not able to find the binaries for the node new_tello_driver_node.py in the current workspace overlay.

Try:

1. Building the ROS workspace again
2. Set the permissions for the new_tello_driver_node.py node by running:
`chmod +x new_tello_driver_node.py`
3. Source the workspace

Then try to launch the launch file again.
